### DISCUSSION TITLE
How to construct a `ratio` from `float`?

### DISCUSSION THREAD

**Discussion by duskmoon314** [typst/typst#3902] [Q&A] [answered]
I'm trying to make a progress bar using `gradient`. But I cannot figure out how to construct a `ratio` from an expression that gives `float` as output.

`ratio(xxx / yyy)` leads to error: ratio has no constructor

**Comment by andreasKroepelin** [comment, marked as answer]
Try multiplying with "one", like `xxx / yyy * 100%` or `xxx / yyy * 1fr`.
